stable/kubernetes-dashboard/templates/clusterrole-readonly.yaml

```diff
@@ -0,0 +1,160 @@
+{{- if and .Values.rbac.create (not .Values.rbac.clusterAdminRole) }}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  labels:
+    app: {{ template "kubernetes-dashboard.name" . }}
+    chart: {{ template "kubernetes-dashboard.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: "{{ template "kubernetes-dashboard.fullname" . }}-readonly"
+  namespace: {{ .Release.Namespace }}
+rules:
+  # Allow Dashboard to get, update and delete Dashboard exclusive secrets.
+  - apiGroups:
+      - ""
+    resources:
+      - secrets
+    resourceNames:
+      - kubernetes-dashboard-key-holder
+      - {{ template "kubernetes-dashboard.fullname" . }}
+    verbs:
+      - get
+      - update
+      - delete
+
+  # Allow Dashboard to get and update 'kubernetes-dashboard-settings' config map.
+  - apiGroups:
+      - ""
+    resources:
+      - configmaps
+    resourceNames:
+      - kubernetes-dashboard-settings
+    verbs:
+      - get
+      - update
+
+  - apiGroups:
+      - ""
+    resources:
+      - configmaps
+      - endpoints
+      - persistentvolumeclaims
+      - pods
+      - replicationcontrollers
+      - replicationcontrollers/scale
+      - serviceaccounts
+      - services
+      - nodes
+      - persistentvolumeclaims
+      - persistentvolumes
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - ""
+    resources:
+      - bindings
+      - events
+      - limitranges
+      - namespaces/status
+      - pods/log
```

Comment:
Ideally nothing bad would ever be logged... stuff happens though. this one also should be treated with extreme care.

Reply:
Hmm.. usually devs want to use the dashboard to check their apps including logs. Of course there should be some centralized system added to the scope but I don't know, maybe you're right. I believe nobody should use the chart _as is_ in production (I don't at least) so people should understand that even read-only mode can cause sensitive data leaks.

Maybe I should add to the PR description or even to the Readme the idea of usage. I suppose that the dashboard is still covered by a proxy with additional authentication layer even though the role is set. And of course in real world people should create more granulated permissions for the team so this one is still for administrators.

So the basic goal of the read-only role comparing to the full admin access is not to hide all the secrets but more to avoid accidental changes in the cluster behind the CI/CD.

Reply:
I'll add the last phrase to `values.yaml`.